[cmake] set 'mcpu=generic' as the default build flag for mkldnn on aarch64

snadampal opened the pull request:
This is to remove the dependencies on mkldnn cmake default definitions

Fixes https://github.com/pytorch/pytorch/issues/109312


cc @gujinghui @PenghuiCheng @XiaobingSuper @jianyuh @jgong5 @mingfeima @sanchitintel @ashokei @jingxu10 @min-jean-cho @yanbing-j @Guobing-Chen @Xia-Weiwen

malfet reviewed (approved):
Would that affect both M1 and aarch64 builds?

snadampal commented:
I think this config is for both M1 and aarch64. But my understanding is, it shouldn't be different from how it is being done in oneDNN makefile today: https://github.com/oneapi-src/oneDNN/pull/1731

snadampal committed:
[cmake] set 'mcpu=generic' as the default build flag for mkldnn on aarch64

This is to fix: https://github.com/pytorch/pytorch/issues/109312
and remove dependencies on mkldnn cmake default definitions

snadampal commented:
@pytorchbot merge
